x/epochs/spec/README.md

```diff
@@ -0,0 +1,45 @@
+<!--
+order: 0
+title: "Epochs Overview"
+parent:
+  title: "epochs"
+-->
+
+# `epochs`
+
+## Abstract
+
+The purpose of `epochs` module is to provide generalized epoch interface to other modules so that they can easily implement epochs without keeping own code for epochs.
+
```

Comment:
Can we add a section talking about what information is specified in an epoch, and whether epoch's get 'block time drift'. 

This block time drift is possible in the current implementation as follows. Say we have an epoch of 100 units, that ends at t=100. We have a block at t=97, and a block at t=104. Currently the code sets the next epoch to begin at t=104, not t=100. I think we should be setting the next epoch to start at t=100.

Reply:
Added thoughts

Reply:
Thanks!